jaypeasee/overlook-hotel
# Refactor: Miscellaneous Bugs

# Pull Request Template

#### What’s this PR do?
This is the final branch/pull request before turning in the project, so there are lots of different little refactors. They include:
1. installs the `moment` `npm dependency` to deal with date consistencies and easier formatting.
1. Moves most sample data each test uses into a sample data file that exports it in arrays. This cleans up the individual test files quite a bit.
1. Cleans up semi-colon and single quotation inconsistencies
1. Updates manager and guest info on the navs asynchronously after posting and deleting data.
1. Refactors method that calculates how much guest has spent on rooms to do correct math.
1. Runs a linter to check for warnings and errors that are fixable.

#### Where should the reviewer start?
* Review the class files and `index.js` to see how `moment` is used to standardize accurate dates.
* Review the class test files and `sample-test-data.js` to see how the sample data is now organized.
* Review the `calculateTotalSpent` method on line 19 of `Guest.js` to see how that has been updated.

#### How should this be manually tested?
* Run `npm test` to make sure all tests are still passing.
* Run `npm start` and go through the entire page and book/delete things on all sorts of dates for both guests and managers. Notice how the nav updates when these changes happen.
* Run `npm run lint` to see what linting warnings remain and see if there is actually anything that should be done about them

#### Any background context you want to provide?
This is likely to be the very last PR before the project is turned in.

#### What are the relevant tickets?
* #60 
* #59 
* #58 
* #17 
* #54 
* #47
